pkg/gofr/config/godotenv.go

```diff
@@ -38,33 +41,28 @@ func (e *EnvLoader) read(folder string) {
 
 	err := godotenv.Load(defaultFile)
 	if err != nil {
-		e.logger.Warnf("Failed to load config from file: %v, Err: %v", defaultFile, err)
+		if errors.Is(err, fs.ErrNotExist) {
+			e.logger.Warnf("Failed to load config from file: %v, Err: %v", defaultFile, err)
+		} else {
+			e.logger.Fatalf("Failed to load config from file: %v, Err: %v", defaultFile, err)
+		}
 	} else {
 		e.logger.Infof("Loaded config from file: %v", defaultFile)
 	}
 
-	switch env {
-	case "":
-		// If 'APP_ENV' is not set, then GoFr will read '.env' from configs directory, and then it will be overwritten
-		// by configs present in file '.local.env'
-		err = godotenv.Overload(overrideFile)
-		if err != nil {
-			e.logger.Debugf("Failed to load config from file: %v, Err: %v", overrideFile, err)
-		} else {
-			e.logger.Infof("Loaded config from file: %v", overrideFile)
-		}
-
-	default:
+	if env != "" {
 		// If 'APP_ENV' is set to x, then GoFr will read '.env' from configs directory, and then it will be overwritten
 		// by configs present in file '.x.env'
 		overrideFile = fmt.Sprintf("%s/.%s.env", folder, env)
+	}
 
-		err = godotenv.Overload(overrideFile)
-		if err != nil {
-			e.logger.Warnf("Failed to load config from file: %v, Err: %v", overrideFile, err)
-		} else {
-			e.logger.Infof("Loaded config from file: %v", overrideFile)
+	err = godotenv.Overload(overrideFile)
+	if err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			e.logger.Fatalf("Failed to load config from file: %v, Err: %v", overrideFile, err)
 		}
+	} else {
```

Comment:
yes, this is `intentional`.. as if you see in line: 61 we are checking if err is not `fs.ErrNotExists` then we are logging err at fatal level.. so, we are skipping the file not exists error and since these files are optional.. hence the removed log.

Reply:
👍

Thanks for confirming you did it on purpose